src/protocol.ts

```diff
@@ -23,6 +22,18 @@ export interface Endpoint {
     listener: EventListenerOrEventListenerObject,
     options?: {}
   ): void;
+}
+
+export interface WindowLike extends EventSource {
```

Comment:
I think the name is a bit misleading. I think something banal but expressive like `PostMessageWithOrigin` would work better.

Reply:
I was a bit torn on the naming. 
Since this type is only used with `windowEndpoint`, I opted for a name that closely resembles the intended type.
But I am open to change it to `PostMessageWithOrigin`.